### DISCUSSION TITLE
Consider authorization for OpenAPI request examples

### DISCUSSION THREAD

**Discussion by arredond** [GitbookIO/community#606] [4. Open API] [answered]
It's very nice that Gitbook takes [OpenAPI security schemes](https://swagger.io/docs/specification/authentication/) into account and shows them in OpenAPI blocks, but the code examples don't, which may cause significant confusion among users.

In this screenshot, I'd expect the code example to include something like `-H "Authorization: Bearer $API_TOKEN"`.

<img width="758" alt="Screenshot 2024-05-06 at 19 48 08" src="https://github.com/GitbookIO/community/assets/19406854/6bf88010-bb69-4bf9-ab58-2c7707a9090f">

Ideally, the "Try it" module would also have the specified authorization method pre-checked, though I recognize this is harder since you use an external provider for it.

<img width="639" alt="Screenshot 2024-05-06 at 19 51 15" src="https://github.com/GitbookIO/community/assets/19406854/cc8af85b-900a-45a6-bebc-639fcb1a625c">

**Comment by addisonschultz** [comment]
Nice suggestions - I'll relay to the team!

**Reply by arredond** [reply]
Any updates on this one? This remains as an odd sticking-point in our docs -- the OpenAPI code block clearly shows that `BearerAuth` is required as the only auth option, but the example Request code doesn't include it in the headers, which makes it much less useful (and even somewhat confusing) for our users

<img width="783" alt="image" src="https://github.com/user-attachments/assets/b9480bca-9b3e-4dca-982c-61948521532b">

**Reply by addisonschultz** [reply]
We've still got an open ticket for this - trying to do a bit debugging and for more context on our side when we pick this up, I believe it's coming from this: https://github.com/GitbookIO/gitbook/blob/main/packages/react-openapi/src/OpenAPICodeSample.tsx#L134

Thanks for your patience on this one so far

**Comment by nolannbiron** [comment, marked as answer]
Hey @arredond, sorry for the delay, here's what I found 

In your specification, it looks like both paths you provided in your screenshot don't have the `security` property. Is this intentional?  

If you add the `security` property like this inside your paths, it will work with our updated OpenAPI blocks.

```json
"security": [
    {
        "bearerAuth": []
    }
]
```

Some of your endpoints already have the `security` property (see [example](https://developers.felt.com/rest-api/api-reference/projects#api-v2-projects-1))

For the “Try it” feature, it displays available `securitySchemes` (even if they are not required for this endpoint). This is normal for testing the endpoint, but we won’t add it inside the block, as it could be misleading.

Let me know if you run into any other issues, happy to help!
